### PR TITLE
Prevent modification of translations

### LINKED ISSUE #7195: Prevent changes on the translations
**Developer should never touch the files managed by Weblate, except during a Weblate sync**

When developers update the translations, it can lead to conflict when Weblate wants to create a PR to push the translation to the project.
We should prevent that.

- [x] Add a check in Danger (error level), to ensure that no file with the format `/values-*/strings.xml` is modified by the developer. An exception can be set up for PR created by Weblate. (Alternatively we could use a GitHub action as per the React SDK https://github.com/matrix-org/matrix-react-sdk/blob/develop/.github/workflows/i18n_check.yml#L12-L29)
- [x] Update the doc https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#internationalisation to mention `id` renaming. A TODO has to be added in the main `strings.xml` file, for the change to be done during Weblate sync.

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Fixes #7195 

## Motivation and context

Avoid being blocked with merge conflicts when doing a Weblate sync.
Update the doc.

## Screenshots / GIFs

NA

## Tests

Running locally against various PRs, using `bundle exec danger pr PR_URL --dangerfile=./tools/danger/dangerfile.js`

### PR with modification of translations

Will throw an error.
For instance #7084, output:
```
## Failures
Some translation files have been edited. Only user `RiotTranslateBot` (i.e. translations coming from Weblate) is allowed to do that.
Please read more about translations management [in the doc](https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#internationalisation).
```

### PR with modification of values/strings.xml

#6608, no error on output

### PR from Weblate

#7201, no error on output

## Tested devices

NA

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by bmarty** [comment]
Adding manually @mnaturel as a reviewer :).

File: tools/danger/dangerfile.js [hunk 1]
```diff
@@ -118,3 +118,10 @@ if (hasPngs) {
 if (github.requested_reviewers.users.length == 0 && !pr.draft) {
     warn("Please add a reviewer to your PR.")
 }
+
+// Check that translations have not been modified by developers
+if (user != "RiotTranslateBot") {
```
Comment (ouchadam) [inline tools/danger/dangerfile.js:123]: :100:

**Review by ouchadam — APPROVED** [review]
great action :100:

**Review by mnaturel — APPROVED** [review]
Thanks for clarifying the process in the doc!

**Pull request merged**